google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/BigtableBatchingCallSettings.java

```diff
@@ -145,9 +194,135 @@ public Builder setRetrySettings(@Nonnull RetrySettings retrySettings) {
       return this;
     }
 
+    /**
+     * Enables / disables latency based throttling. If enabling the setting, targetRpcLatency needs
+     * to be set.
+     */
+    public Builder setLatencyBasedThrottling(
+        boolean isLatencyBasedThrottlingEnabled, @Nullable Long targetRpcLatency) {
+      Preconditions.checkArgument(
+          !isLatencyBasedThrottlingEnabled || targetRpcLatency != null,
+          "target RPC latency must be set if latency based throttling is enabled");
+      Preconditions.checkArgument(
+          targetRpcLatency == null || targetRpcLatency > 0,
+          "if target RPC latency is set, it must be greater than 0");
+      this.isLatencyBasedThrottlingEnabled = isLatencyBasedThrottlingEnabled;
+      this.targetRpcLatencyMs = isLatencyBasedThrottlingEnabled ? targetRpcLatency : null;
+      return this;
+    }
+
+    /** Gets target rpc latency if latency based throttling is enabled. Otherwise returns null. */
+    @Nullable
+    public Long getTargetRpcLatencyMs() {
+      return isLatencyBasedThrottlingEnabled ? targetRpcLatencyMs : null;
+    }
+
+    /** Gets if latency based throttling is enabled. */
+    public boolean isLatencyBasedThrottlingEnabled() {
+      return this.isLatencyBasedThrottlingEnabled;
+    }
+
+    /**
+     * Gets the {@link DynamicFlowControlSettings} that'll be used to set up a {@link
+     * FlowController} for throttling.
+     *
+     * <p>By default, this will allow a maximum of 1000 entries per channel of {@link
```

Comment:
I'm not sure if the defaults should be set/documented here. This class can be used for both reads and writes. I'm not sure that the 2 should share their defaults. Can we document this on the BigtableDataSettings or StubSettings?

Reply:
I thought this class is only for BulkMutation? `public final class BigtableBatchingCallSettings extends UnaryCallSettings<BulkMutation, Void>` and reads uses `BigtableBulkReadRowsCallSettings`.

I put the documentation here because: default `DynamicFlowControlSettings` depends on the default settings for `maxElementCount` and `maxRequestBytes`.  And nowhere else is `DynamicFlowControlSettings` mentioned. So I thought it might be easier to put all the explanations here?

Reply:
naming is hard :(
BigtableBatchingCallSettings  sounds generic and I forgot that we created a separate class for reads